src/test/kotlin/org/virtuslab/pulumikotlin/codegen/step2intermediate/PulumiNameTest.kt

```diff
@@ -304,8 +398,13 @@ internal class PulumiNameTest {
 
         // then
         assertAll(
-            { assertEquals("overrideProvider", pulumiName.providerName) },
-            { assertEquals(listOf("org", "example", "overrideProvider", "overrideModule"), pulumiName.namespace) },
+            { assertEquals("provider", pulumiName.providerName) },
+            {
+                assertEquals(
+                    listOf("org", "example", "overrideProvider", "overrideModule"),
+                    pulumiName.namespace,
+                )
+            },
             { assertEquals("ObjectName", pulumiName.name) },
```

Comment:
(not really about this PR and nothing we can do now?)

I'm not sure if we should have such tests. We should test the public interface (e.g. `.toPackage`, `toFunctionGroupObjectName`) and I'm not sure `providerName`, `namespace` and `name` should be public fields.

Reply:
True, I'll create a task 👍

Reply:
https://github.com/VirtuslabRnD/pulumi-kotlin/issues/273